libs/auth/src/inserted_smart_card_auth.ts

```diff
@@ -342,21 +348,33 @@ export class InsertedSmartCardAuth implements InsertedSmartCardAuthApi {
               assert(cardDetails !== undefined);
               const { user } = cardDetails;
               if (currentAuthStatus.status === 'logged_out') {
-                if (user.role === 'system_administrator') {
-                  return isFeatureFlagEnabled(
-                    BooleanEnvironmentVariableName.SKIP_PIN_ENTRY
-                  )
-                    ? { status: 'logged_in', user }
-                    : { status: 'checking_pin', user };
-                }
-                if (user.role === 'election_manager') {
-                  return isFeatureFlagEnabled(
-                    BooleanEnvironmentVariableName.SKIP_PIN_ENTRY
-                  )
-                    ? { status: 'logged_in', user }
-                    : { status: 'checking_pin', user };
+                const skipPinEntry = isFeatureFlagEnabled(
+                  BooleanEnvironmentVariableName.SKIP_PIN_ENTRY
+                );
+                switch (user.role) {
+                  case 'system_administrator': {
+                    return skipPinEntry
+                      ? { status: 'logged_in', user }
+                      : { status: 'checking_pin', user };
+                  }
+                  case 'election_manager': {
+                    return skipPinEntry
+                      ? { status: 'logged_in', user }
+                      : { status: 'checking_pin', user };
+                  }
```

Comment:
I know it looks like some of these cases could be merged, but TS requires me to separate them out